### PR TITLE
feat: Add option for a custom image and asset loader

### LINKED ISSUE #2568: Tiled asset loader
# Problem to solve

I want to give users in my app the option to create their own maps. For this I want to sandbox the asset loading process.

# Proposal
Create an asset loader argument, for example with `(String path) => Uint8List` to get the image.

# More information

This feature can also has the feature to create multiple asset directories for different levels.

### PR DESCRIPTION
**Pull request by CodeDoctorDE**
# Description

Add images and bundle to the named parameters for the creation of a tiled map.

## Checklist

- [x] I have followed the [Contributor Guide] when preparing my PR.
- [ ] I have updated/added tests for ALL new/updated/fixed functionality.
- [ ] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [ ] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

No

### Migration instructions

\-

## Related Issues

Closes #2568

### CODE REVIEW

**Comment by CodeDoctorDE** [comment]
I don't know why the ci fails
> Test run was interrupted.
Error: The operation was canceled.

**Comment by spydon** [comment]
Sometimes the test stage gets cancelled by the CI for unknown reasons, this is what happened here.
What is strange is that the analyze step passed, because there are a few analyze issues, one line that is too long and some missing trailing commas (you can see the issues annotated in the `files` tab in GitHub).

File: packages/flame_tiled/lib/src/flame_tsx_provider.dart [hunk 1]
```diff
@@ -5,7 +5,8 @@ import 'package:xml/xml.dart';
 
 /// A implementation of [TsxProvider] use by RenderableTileMap.
 ///
-/// It uses [Flame.bundle] or a custom asset bundle and has a built-in cache for the file read.
+/// It uses [Flame.bundle] or a custom asset bundle
+/// and has a built-in cache for the file read.
```
Comment (spydon) [inline packages/flame_tiled/lib/src/flame_tsx_provider.dart:9]: This is fine, but ideally it should only break once it reaches 80 and not before.

Comment (CodeDoctorDE) [reply packages/flame_tiled/lib/src/flame_tsx_provider.dart:9]: Hmm, okay. Should I change that?
For me it makes sense to break if a new part of the sentence has started

Comment (spydon) [reply packages/flame_tiled/lib/src/flame_tsx_provider.dart:9]: It's a bit like using a line break as a comma, something one wouldn't do in a book for example.
Anyhow this is fine, but now you know until the next PR. :)

**Review by spydon — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**